messaging/src/main/java/org/axonframework/deadline/jobrunr/JobRunrDeadlineManager.java

```diff
@@ -109,13 +109,12 @@ public String schedule(@Nonnull Instant triggerDateTime, @Nonnull String deadlin
                            @Nullable Object messageOrPayload,
                            @Nonnull ScopeDescriptor deadlineScope) {
         DeadlineMessage<Object> deadlineMessage = asDeadlineMessage(deadlineName, messageOrPayload, triggerDateTime);
-        UUID deadlineId = UUID.fromString(deadlineMessage.getIdentifier());
+        UUID deadlineId = UUID.randomUUID();
```

Comment:
It's a nit, but worth adjusting: we should use the `IdentifierFactory` from Axon Framework here right now.
All `Message` implementations, if the user doesn't provide the identifier itself, default to the `IdentifierFactory` (that can be adjusted by the users through the Service Loader mechanism).

As we're not constructing the ID ourselves, it's fair to default to this behavior here too.

Reply:
That returns a `String` through, while JobRunr needs a `UUID`. In the case of the `DefaultIdentifierFactory` it's a `String` version of a `UUID`. But the reason for the change was to make sure it works when someone uses a `IdentifierFactory` that doesn't return a `UUID`. Optionally we could add a `UuidFactory` to give users more control over the uuid's if thay want, but I'm not sure that's needed.

Reply:
Hmm, alright, that's a shame. 
But, that also means we have a very good reason why to stick to `UUID`: because it's required by JobRunr.

Knowing that I think we're good here. 
Thanks for the information, @gklijs!